internal/operators/metallb/metallb_operator.go

```diff
@@ -0,0 +1,194 @@
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID()}, nil
+}
+
+// GenerateManifests generates manifests for the operator
+func (o *operator) GenerateManifests(cluster *common.Cluster) (map[string][]byte, []byte, error) {
+	return Manifests(cluster)
+}
+
+// GetProperties provides description of operator properties: none required
+func (o *operator) GetProperties() models.OperatorProperties {
+	return models.OperatorProperties{}
+}
+
+// GetMonitoredOperator returns MonitoredOperator corresponding to the LSO
+func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
+	return &Operator
+}
+
+// GetHostRequirements provides operator's requirements towards the host
+func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster, _ *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	log := logutil.FromContext(ctx, o.log)
+	preflightRequirements, err := o.GetPreflightRequirements(ctx, cluster)
+	if err != nil {
+		log.WithError(err).Errorf("Cannot retrieve preflight requirements for cluster %s", cluster.ID)
+		return nil, err
+	}
+	return preflightRequirements.Requirements.Master.Quantitative, nil
+}
+
+// GetPreflightRequirements returns operator hardware requirements that can be determined with cluster data only
+func (o *operator) GetPreflightRequirements(context context.Context, cluster *common.Cluster) (*models.OperatorHardwareRequirements, error) {
+	dependecies, err := o.GetDependencies(cluster)
+	if err != nil {
+		return &models.OperatorHardwareRequirements{}, err
+	}
+	return &models.OperatorHardwareRequirements{
+		OperatorName: o.GetName(),
+		Dependencies: dependecies,
+		Requirements: &models.HostTypeHardwareRequirementsWrapper{
+			Master: &models.HostTypeHardwareRequirements{
+				Quantitative: &models.ClusterHostRequirementsDetails{
+					CPUCores: o.config.MetalLBCPUPerHost,
+					RAMMib:   o.config.MetalLBMemoryPerHostMiB,
+				},
+			},
+			Worker: &models.HostTypeHardwareRequirements{
+				Quantitative: &models.ClusterHostRequirementsDetails{},
+			},
+		},
+	}, nil
+}
+
+func (o *operator) GetSupportedArchitectures() []string {
+	return []string{common.X86CPUArchitecture, common.ARM64CPUArchitecture}
+}
+
+// parse and validate the operator's properties field
+func parsePropertiesField(cluster *common.Cluster) (Properties, error) {
+	propertiesObj := Properties{}
+	properties := ""
+
+	for _, mo := range cluster.MonitoredOperators {
+		if mo != nil && mo.Name == Operator.Name {
+			properties = mo.Properties
+			break
+		}
+	}
+
+	if properties == "" {
+		return propertiesObj, fmt.Errorf("Properties field for operator %s can not be empty", Operator.Name)
+	}
+
+	err := json.Unmarshal([]byte(properties), &propertiesObj)
+	if err != nil {
+		return propertiesObj, err
+	}
+
+	ip := net.ParseIP(propertiesObj.ApiIP)
```

Comment:
From manifests we support only ipv4.  Should not we validate that ip is ipv4?

Reply:
@flaper87?

Reply:
@flaper87 @tsorya I left the code is. I think that the code should not validate type of IP. If our manifests will support IPv6 in the future then no changes will be required here. If IPv6 is not supported then there should be a location, earlier in time, that validates it.

Reply:
Just for me to understand. This ip can be any ip right or host should have this subnet configured? For example why adding ipv6 ip will fail earlier? Where do we verify it?

Reply:
@tsorya  any ip that metallb operator can use is ok. so far I did not find any thing that prevents metallb for answering any ip on any interface (NIC/device).
You wrote that 'From manifests we support only ipv4'. Who limits that? if no one, then this limitation does not exist. If someone somewhere limits that then this is where the validation is performed. I think we should not add validations. It only makes the code more sensitive to coding mistakes and changes in the future. But your call

Reply:
The idea of assisted is actually add validation in order to make user life easier. We try to validate as maximum as we can actually. Currently you add ip as {ip}/32 that makes it valid only for ipv4. I think that we need or to change code to support both types of ips or to validate that ip is valid.

Reply:
@tsorya @flaper87 done

Reply:
can you use https://github.com/openshift/assisted-service/blob/master/internal/network/family_utils.go#L8 ? just to make less code

Reply:
@tsorya really? less code? what are we saving? a line? and instead of using a builtin, maintained, standard library use our implementation?

Reply:
Lets say we already have it and use it all over the code, why not to use it? 
 if i understand right it will be changed to:
`if !IsIPv4Addr(propertiesObj.ApiIP) {
   return propertiesObj, fmt.Errorf("MetalLB API IP is not a valid IPv4 address")
}
`
It is less enough code as for me and  more readable.

Reply:
done